qmetaobject_impl/src/qobject_impl.rs

```diff
@@ -686,7 +686,7 @@ pub fn generate(input: TokenStream, is_qobject: bool, qt_version: QtVersion) ->
     } else {
         quote! {
             #[allow(unused_variables)]
-            let mut obj = ::std::mem::transmute::<*mut ::std::os::raw::c_void, &mut #name #ty_generics>(o);
+            let mut obj = &*o;
```

Comment:
i would think you still need to cast to the right type explicitly.

Reply:
Yeah this solution looks wrong. I kept the transmute to cast to the right type as `*mut`, but dereferenced it now to `&mut` outside of the transmute.